lumicks/pylake/channel.py

```diff
@@ -21,14 +22,54 @@ class Slice:
         Plot labels: "x", "y", "title".
     calibration: ForceCalibration
     """
-    def __init__(self, data_source, labels=None, calibration=None):
+    def __init__(self, data_source, labels=None, calibration=None, unit=None):
         self._src = data_source
         self.labels = labels or {}
+        self.unit = unit
         self._calibration = calibration
 
     def __len__(self):
         return len(self._src)
 
+    def _unpack_other(self, other):
+        if isinstance(other, self.__class__):
+            return other._src.data
+        else:
+            return other
+
+    def compare_unit(self, other, operation):
+        other_unit = getattr(other, "unit", None)
+        if other_unit:
+            return determine_unit(self.unit, other_unit, operation)
+
+    def __add__(self, other):
+        unit = self.compare_unit(other, 'add')
+        return Slice(self._src._with_data_source(self._src.data + self._unpack_other(other)), unit=unit)
+
+    def __sub__(self, other):
+        unit = self.compare_unit(other, 'sub')
+        return Slice(self._src._with_data_source(self._src.data - self._unpack_other(other)), unit=unit)
+
+    def __truediv__(self, other):
+        unit = self.compare_unit(other, 'div')
+        return Slice(self._src._with_data_source(self._src.data / self._unpack_other(other)), unit=unit)
+
+    def __mul__(self, other):
+        unit = self.compare_unit(other, 'mul')
+        return Slice(self._src._with_data_source(self._src.data * self._unpack_other(other)), unit=unit)
+
+    def __rtruediv__(self, other):
+        unit = self.compare_unit(other, 'div')
+        return Slice(self._src._with_data_source(self._unpack_other(other)/self._src.data), unit=unit)
+
+    def __rsub__(self, other):
+        unit = self.compare_unit(other, 'sub')
+        return Slice(self._src._with_data_source(self._unpack_other(other)-self._src.data), unit=unit)
```

Comment:
I'm not even sure if this is standard or not but `__rsub__` and `__rtruediv__` both lack space between operators:
`self._unpack_other(other)-self._src.data` -> `self._unpack_other(other) - self._src.data`

Reply:
👍 Added